robusta/txsim.go

```diff
@@ -0,0 +1,237 @@
+		return nil, err
+	}
+
+	return &Txsim{
+		Name:       name,
+		Instance:   instance,
+		Version:    version,
+		SignerKey:  signerKey,
+		NetworkKey: networkKey,
+		AccountKey: accountKey,
+
+		RPCEndpoints:  rpcEndpoints,
+		GRPCEndpoints: grpcEndpoints,
+		PollTime:      pollTime,
+		BlobSizes:     blobSizes,
+		Blob:          blob,
+		BlobAmounts:   blobAmounts,
+		Seed:          seed,
+	}, nil
+}
+
+func (ts *Txsim) Init() error {
+	// Initialize file directories
+	rootDir := os.TempDir()
+	nodeDir := filepath.Join(rootDir, ts.Name)
+	for _, dir := range []string{
+		filepath.Join(nodeDir, "config"),
+		filepath.Join(nodeDir, "data"),
+	} {
+		if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+			return fmt.Errorf("error creating directory %s: %w", dir, err)
+		}
+	}
+
+	//// Create and write the config file
+	//cfg, err := MakeConfig(n)
+	//if err != nil {
+	//	return fmt.Errorf("making config: %w", err)
+	//}
+	//configFilePath := filepath.Join(nodeDir, "config", "config.toml")
+	//config.WriteConfigFile(configFilePath, cfg)
+	//
+	//// Store the genesis file
+	//genesisFilePath := filepath.Join(nodeDir, "config", "genesis.json")
+	//err = genesis.SaveAs(genesisFilePath)
+	//if err != nil {
+	//	return fmt.Errorf("saving genesis: %w", err)
+	//}
+	//
+	//// Create the app.toml file
+	//appConfig, err := MakeAppConfig(n)
+	//if err != nil {
+	//	return fmt.Errorf("making app config: %w", err)
+	//}
+	//appConfigFilePath := filepath.Join(nodeDir, "config", "app.toml")
+	//serverconfig.WriteConfigFile(appConfigFilePath, appConfig)
+
+	// Store the node key for the p2p handshake
+	nodeKeyFilePath := filepath.Join(nodeDir, "config", "node_key.json")
+	err := (&p2p.NodeKey{PrivKey: ts.NetworkKey}).SaveAs(nodeKeyFilePath)
+	if err != nil {
+		return err
+	}
+
+	err = os.Chmod(nodeKeyFilePath, 0o777)
+	if err != nil {
+		return fmt.Errorf("chmod node key: %w", err)
+	}
+
+	// Store the validator signer key for consensus
+	pvKeyPath := filepath.Join(nodeDir, "config", "priv_validator_key.json")
+	pvStatePath := filepath.Join(nodeDir, "data", "priv_validator_state.json")
+	(privval.NewFilePV(ts.SignerKey, pvKeyPath, pvStatePath)).Save()
+
+	//addrBookFile := filepath.Join(nodeDir, "config", "addrbook.json")
+	//err = WriteAddressBook(peers, addrBookFile)
+	//if err != nil {
+	//	return fmt.Errorf("writing address book: %w", err)
+	//}
+
+	//err = n.Instance.AddFile(configFilePath, filepath.Join(remoteRootDir, "config", "config.toml"), "10001:10001")
+	//if err != nil {
+	//	return fmt.Errorf("adding config file: %w", err)
+	//}
+	//
+	//err = n.Instance.AddFile(genesisFilePath, filepath.Join(remoteRootDir, "config", "genesis.json"), "10001:10001")
+	//if err != nil {
+	//	return fmt.Errorf("adding genesis file: %w", err)
+	//}
+	//
+	//err = n.Instance.AddFile(appConfigFilePath, filepath.Join(remoteRootDir, "config", "app.toml"), "10001:10001")
+	//if err != nil {
+	//	return fmt.Errorf("adding app config file: %w", err)
+	//}
```

Comment:
why is this commented out?

Reply:
it's a copy&paste leftover, thanks for mentioning it